addons/website_sale/tests/test_website_sale_product_template.py

```diff
@@ -45,3 +45,15 @@ def test_website_sale_get_additional_configurator_data(self):
 
         self.assertEqual(configurator_data['category_name'], "Test category")
         self.assertEqual(configurator_data['currency_name'], 'EUR')
+
+    def test_archived_product_erased_form_sol(self):
+        self.product.action_archive()
+        for prod in self.cart.order_line:
+            self.assertTrue(prod.product_id != self.product, "archived product need to be erased form cart")
```

Comment:
`assertNotIn` is the more appropriate assert for this case:
```suggestion
        self.assertNotIn(
            self.product, self.cart.order_line.product_id,
            "Archived product should be deleted from the cart",
        )
```
Also try to document tests in the form of "X *should* Y" or "Y *shouldn't* Z"